docs/env-providers.md

```diff
@@ -0,0 +1,51 @@
+# Development and Testing Environment Providers
+
+All following providers allow a common workflow:
+
+ * `cluster/up.sh` to create the environment
+ * `cluster/down.sh` to stop the environment
+ * `cluster/sync.sh` to build the code and deploy it
```

Comment:
can break out the build and deploy steps?  It's common for me to want to sync builds to my vagrant env and manually upgrade a single component without redeploying everything.

Reply:
Good point. How exactly are you using it, to validate that I am not destrohing that flow? cluster/deploy.sh is still there. Added a line here too. It just works for every provider, since it only requires _kubectl from the provider. That is why it is not mentioned here.

Reply:
This is my flow. 

- write some code for a kubevirt component
- call 'make vagrant-sync-build'
- manually cycle the single component i was working on.

it would be nice if i could sync just a single container, but that functionality doesn't exist now and I don't want to add more to this pr.